### PR TITLE
Add password support and SFTP_PASSWORD env var

### PR DESCRIPTION
**Pull request by yegortimoshenko**
For some SFTP servers, especially those that can't be configured, the only way to authenticate is by using a passsword. This pull request adds optional password support to all handlers.

### CODE REVIEW

**Comment by lqueryvg** [comment]
@yegortimoshenko
I'm happy to merge this.
Can you please update your branch and push again.
For some reason the status checks aren't running. And I don't see to have an Update Branch button.

**Review by lqueryvg — APPROVED** [review]

**Comment by lqueryvg** [comment]
I'm going to try closing and re-opening the PR.
To see if that kicks the status checks back into life.

**Comment by yegortimoshenko** [comment]
Looks like reopening the PR worked :) I see the test failures and I'll fix them sometime today!

**Comment by lqueryvg** [comment]
@yegortimoshenko, looks like I got the checks working !

FYI, you should be able to run the following sequence locally.
I just did it on master using node v12.18.4.

```
yarn clean
yarn
yarn lint
yarn test
yarn test:ci
```

**Comment by lqueryvg** [comment]
> Looks like reopening the PR worked :) I see the test failures and I'll fix them sometime today!

I also had to modify the Github Actions workflow to add the pull_request trigger 😉

File: handlers/lib/config.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
   SFTP_SOURCE_S3_REGEXP_STRIP      push,retry      optional
-  SFTP_HOST                        all             
-  SFTP_PORT                        all             
-  SFTP_USER                        all             
-  SFTP_PRIVATE_KEY                 all             
+  SFTP_HOST                        all
+  SFTP_PORT                        all
+  SFTP_USER                        all
+  SFTP_PASSWORD                    all             optional
+  SFTP_PRIVATE_KEY                 all             optional
```
Comment (yegortimoshenko) [inline handlers/lib/config.js:11]: I've set both of these variables as `optional`, even though one of them typically has to be set. New code in `envVar` should handle this. Let me know if this approach should be changed.

File: handlers/lib/config.js [hunk 2]
```diff
@@ -61,12 +62,23 @@ const initEnvVars = ftpActionType => {
   }
 };
 
-const getEnv = varName => {
+const getEnv = (varName, altVars = []) => {
```
Comment (yegortimoshenko) [inline handlers/lib/config.js:65]: `altVars` here specifies a list of alternative environment variables that can be set in lieu of `varName`.

I believe it's technically possible to set up a non-authenticated SFTP endpoint, in which case this code isn't needed, but support for non-authenticated SFTP will come in exchange of slightly worse error messages for typical use cases.

File: handlers/lib/sshConfig.js [hunk 3]
```diff
@@ -6,7 +6,8 @@ module.exports.getSSHConfig = () => {
     host: getEnv("SFTP_HOST"),
     port: getEnv("SFTP_PORT"),
     username: getEnv("SFTP_USER"),
-    privateKey: getEnv("SFTP_PRIVATE_KEY"),
+    password: getEnv("SFTP_PASSWORD", ["SFTP_PRIVATE_KEY"]),
+    privateKey: getEnv("SFTP_PRIVATE_KEY", ["SFTP_PASSWORD"]),
```
Comment (yegortimoshenko) [inline handlers/lib/sshConfig.js:10]: I believe it's not an issue if *both* password and SSH key is set, in that case what's used depends on the server.

File: handlers/pull.test.js [hunk 4]
```diff
@@ -48,6 +48,18 @@ describe("pull", () => {
...
+    delete process.env.SFTP_PRIVATE_KEY;
+    try {
+      await pull();
+    } catch (err) {
+      expect(err.message).toMatch(
+        /Environment variables .* not set, need at least one/
+      );
+    }
+  });
+
```
Comment (yegortimoshenko) [inline handlers/pull.test.js:62]: This is a test for a new error message that appears if none of the mutually substitutable environment variables are set.

**Review by yegortimoshenko — COMMENTED** [review]
Tests now pass and coverage is back to 100%! Below is some rationale for the changes.

Let me know if you want to change/tweak anything!

**Comment by yegortimoshenko** [comment]
Thanks for the merge :green_heart:

**Pull request merged**